impl/authorization/authorization_test.go

```diff
@@ -85,101 +89,105 @@ func setup() *authz {
-			"1",
-			"2",
-			"",
-			authzpb.Permission_LOG,
-			true,
+			description: "other accessing profile, authorized second resource",
+			ctx:         authentication.WithOutgoingFakeAuth(ctx, admin3),
+			domainID:    "1",
+			appID:       "2",
+			userID:      "",
+			permission:  authzpb.Permission_LOG,
 		},
 		{
-			"not authorized, no resource label",
-			authentication.NewSecurityContext(admin1),
-			"1",
-			"10",
-			"",
-			authzpb.Permission_WRITE,
-			false,
+			description: "not authorized, no resource label",
+			ctx:         authentication.WithOutgoingFakeAuth(ctx, admin1),
+			domainID:    "1",
+			appID:       "10",
+			userID:      "",
+			permission:  authzpb.Permission_WRITE,
+			wantCode:    codes.PermissionDenied,
 		},
 		{
-			"not authorized, no label_to_role defined",
-			authentication.NewSecurityContext(admin1),
-			"1",
-			"4",
-			"",
-			authzpb.Permission_LOG,
-			false,
+			description: "not authorized, no label_to_role defined",
+			ctx:         authentication.WithOutgoingFakeAuth(ctx, admin1),
+			domainID:    "1",
+			appID:       "4",
+			userID:      "",
+			permission:  authzpb.Permission_LOG,
+			wantCode:    codes.PermissionDenied,
 		},
 		{
-			"not authorized, empty role definition",
-			authentication.NewSecurityContext(admin1),
-			"1",
-			"3",
-			"",
-			authzpb.Permission_WRITE,
-			false,
+			description: "not authorized, empty role definition",
+			ctx:         authentication.WithOutgoingFakeAuth(ctx, admin1),
+			domainID:    "1",
+			appID:       "3",
+			userID:      "",
+			permission:  authzpb.Permission_WRITE,
+			wantCode:    codes.PermissionDenied,
 		},
 		{
-			"not authorized, wrong permission",
-			authentication.NewSecurityContext(admin2),
-			"1",
-			"1",
-			"",
-			authzpb.Permission_WRITE,
-			false,
+			description: "not authorized, wrong permission",
+			ctx:         authentication.WithOutgoingFakeAuth(ctx, admin2),
+			domainID:    "1",
+			appID:       "1",
+			userID:      "",
+			permission:  authzpb.Permission_WRITE,
+			wantCode:    codes.PermissionDenied,
 		},
 		{
-			"not authorized principal",
-			authentication.NewSecurityContext(admin4),
-			"1",
-			"1",
-			"",
-			authzpb.Permission_WRITE,
-			false,
+			description: "not authorized principal",
+			ctx:         authentication.WithOutgoingFakeAuth(ctx, admin4),
+			domainID:    "1",
+			appID:       "1",
+			userID:      "",
+			permission:  authzpb.Permission_WRITE,
+			wantCode:    codes.PermissionDenied,
 		},
 	} {
-		err := a.IsAuthorized(tc.sctx, tc.domainID, tc.appID, tc.userID, tc.permission)
-		if got, want := err == nil, tc.success; got != want {
-			t.Errorf("%v: IsAuthorized err == nil: %v, want %v", tc.description, got, want)
+		// Convert outgoing context to incoming context.
+		inCtx := metautils.ExtractOutgoing(tc.ctx).ToIncoming(ctx)
+		sctx, err := authentication.FakeAuthFunc(inCtx)
+		if err != nil {
+			t.Errorf("FakeAuthFunc(): %v", err)
+			continue
```

Comment:
maybe t.FailNow() ?

Reply:
We want to run the rest of the tests :-)
I'll put this in a `t.Run()`

Reply:
r.Run() is better.